autonomy/cli/analyse.py

```diff
@@ -124,37 +124,40 @@ def abci_group() -> None:
     """Analyse ABCI apps of an agent service."""
 
 
-@abci_group.command(name="docstrings")
-@click.argument(
-    "packages_dir",
-    type=click.Path(dir_okay=True, file_okay=False, exists=True),
-    default=Path("packages/"),
-)
-@click.option("--check", is_flag=True, default=False)
-def docstrings(packages_dir: Path, check: bool) -> None:
+@analyse_group.command(name="docstrings")
+@click.option("--update", is_flag=True, default=False)
+@pass_ctx
+def docstrings(ctx: Context, update: bool) -> None:
     """Analyse ABCI docstring definitions."""
 
-    packages_dir = Path(packages_dir)
+    packages_dir = Path(ctx.registry_path).resolve()
     needs_update = set()
     abci_compositions = packages_dir.glob("*/skills/*/rounds.py")
 
-    try:
-        for path in sorted(abci_compositions):
-            click.echo(f"Processing: {path}")
-            if process_module(path, check=check):
-                needs_update.add(str(path))
-
-        if len(needs_update) > 0:
-            file_string = "\n".join(sorted(needs_update))
-            if check:
-                raise click.ClickException(
-                    f"Following files needs updating.\n\n{file_string}"
-                )
-            click.echo(f"\nUpdated following files.\n\n{file_string}")
-        else:
-            click.echo("No update needed.")
-    except Exception as e:  # pylint: disable=broad-except
-        raise click.ClickException(str(e)) from e
+    if packages_dir.name != PACKAGES:
+        raise click.ClickException(
+            f"packages directory {packages_dir} is not named '{PACKAGES}'"
+        )
+
+    with sys_path_patch(packages_dir.parent):
```

Comment:
Same as fsm spec check, we append the packages directory to `sys.paths` temporarily